CONTRIBUTING.md

```diff
@@ -63,6 +63,24 @@ https://github.com/elastic/logstash/labels/low%20hanging%20fruit
 
 Using IntelliJ? See a detailed getting started guide [here](https://docs.google.com/document/d/1kqunARvYMrlfTEOgMpYHig0U-ZqCcMJfhvTtGt09iZg/pub).
 
+## Breaking Changes
+
+When introducing new behaviour, we favor implementing it in a non-breaking way that users can opt into, meaning users' existing configurations continue to work as they expect them to after upgrading Logstash.
+
+But sometimes it is necessary to introduce "breaking changes", whether that is removing an option that doesn't make sense anymore, changing the default value of a setting, or reimplementing a major component differently for performance or safety reasons.
+
+When we do so, we need to acknowledge the work we are placing on the rest of our users the next time they upgrade, and work to ensure that they can upgrade confidently.
+
+Where possible, we:
+ 1. first implement new behaviour in a way that users can explicitly opt into (MINOR),
+ 2. introduce deprecation warnings when old behaviour is used, communicating the pending change (MINOR, potentially along-side #1),
+ 3. change the default to be new behaviour, optionally allowing users to opt out in favor of the old behaviour (MAJOR), and eventually
```

Comment:
at this point we should also note that a entry should be added to the next major's breaking change document https://github.com/elastic/logstash/blob/master/docs/static/breaking-changes.asciidoc, what do you think?

Reply:
```suggestion
 3. change the default to be new behaviour, communicating the breaking change and optionally allowing users to opt out in favor of the old behaviour (MAJOR), and eventually
```

Reply:
Consider  keeping verbs parallel "change," "communicate," and "allow"

Reply:
```suggestion
 3. change the default to be new behaviour, communicate the breaking change, optionally allow users to opt out in favor of the old behaviour (MAJOR), and eventually
```